src/Membership.sol

```diff
@@ -11,432 +11,353 @@ error InvalidRateLimit();
 
 // It's not possible to acquire the rate limit due to exceeding the expected limits
 // even after attempting to erase expired memberships
-error ExceedAvailableMaxRateLimitPerEpoch();
+error ExceededMaxTotalRateLimit();
 
-// This membership is not in grace period yet
+// This membership is not in grace period yet // FIXME: yet or also already?
 error NotInGracePeriod(uint256 idCommitment);
 
 // The sender is not the holder of the membership
-error NotHolder(uint256 idCommitment);
+error AttemptedExtensionByNonHolder(uint256 idCommitment);
 
-// This membership cannot be erased (either it is not expired or not in grace period and/or not the owner)
+// This membership cannot be erased (either it is not expired or not in grace period and/or not the owner) // FIXME:
+// separate into two errors?
 error CantEraseMembership(uint256 idCommitment);
 
 abstract contract MembershipUpgradeable is Initializable {
     using SafeERC20 for IERC20;
 
     /// @notice Address of the Price Calculator used to calculate the price of a new membership
-    IPriceCalculator public priceCalculator;
+    IPriceCalculator public priceCalculator; // FIXME: naming: price vs deposit?
 
-    /// @notice Maximum total rate limit of all memberships in the tree
-    uint32 public maxTotalRateLimitPerEpoch;
+    /// @notice Maximum total rate limit of all memberships in the membership set (messages per epoch)
+    uint32 public maxTotalRateLimit;
 
     /// @notice Maximum rate limit of one membership
-    uint32 public maxRateLimitPerMembership;
+    uint32 public maxMembershipRateLimit;
 
     /// @notice Minimum rate limit of one membership
-    uint32 public minRateLimitPerMembership;
+    uint32 public minMembershipRateLimit;
 
-    /// @notice Membership billing period
-    uint32 public expirationTerm;
+    /// @notice Membership active period duration (A in the spec)
+    uint32 public activeStateDuration;
 
-    /// @notice Membership grace period
-    uint32 public gracePeriod;
+    /// @notice Membership grace period duration (G in the spec)
+    uint32 public gracePeriodDuration;
```

Comment:
Pick a lane between `StateDuration` and `PeriodDuration`

Reply:
I think this has been solved already.